troposphere/static/js/components/common/InstanceHistorySection.jsx

```diff
@@ -35,19 +46,73 @@ var InstanceHistorySection = React.createClass({
         });
     },
 
+    style() {
+        return {
+            refreshIcon: {
+                float: "right",
+                color: "lightgrey"
+            },
+        }
+    },
+
+    renderRefreshButton() {
+        let { refreshing } = this.state;
+        let { refreshIcon } = this.style();
+        let controlsClass = "glyphicon glyphicon-refresh";
+
+        if (refreshing) {
+            controlsClass += " refreshing"
+            refreshIcon.color = "inherit";
+        }
+
+        return (
+        <span className={controlsClass} style={refreshIcon} onClick={this.onRefresh} />
+        );
+    },
+
+    renderHistoryRow: function(historyItem) {
+        let profile = stores.ProfileStore.get();
+        let is_staff_user = (profile) ? profile.get("is_staff") : false;
+        let extra = historyItem.get('extra'),
+            formattedStartDate = moment(historyItem.get("start_date")).format("MMMM Do YYYY, h:mm a"),
+            formattedEndDate = "Present";
+        if (historyItem.get("end_date") && historyItem.get("end_date").isValid()) {
+            formattedEndDate = moment(historyItem.get("end_date")).format("MMMM Do YYYY, h:mm a");
+        }
+        let formattedExtra = "";
+        let formattedExtraLines = [];
+        let show_traceback = (is_staff_user || context.hasEmulatedSession() );
+        if(extra && 'display_error' in extra) {
+            formattedExtra = extra['display_error'];
+            if('traceback' in extra && show_traceback) {
+                formattedExtra = formattedExtra + "\\n" + extra['traceback']
+                formattedExtraLines = formattedExtra.split('\\n');
+            }
```

Comment:
Really, just a simple show / hide component with a nice animation.
Made a note to do